CR-1291: Updating OTM diagrams

PawelWMS opened the pull request:
Updating OTM diagrams to include the OBT in the initial steps of Owership Transfer.

PawelWMS committed:
CR-1291: Updating the OTM diagrams to include OBT in the initial steps of Ownership Transfer.

craigpratt commented on security/DOXM-MFGCERT.txt:
```diff
@@ -2,46 +2,62 @@
-  OT->ND: ClientHello(TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8,cookie)
-  ND->OT: ServerHello(TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8)\nCertificate*\nServerKeyExchange(ECDH PublicKey + ECC Curve Param)\nServerHelloDone()
+  OT->ND: ClientHello(TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8_SHA256,cookie)
+  ND->OT: ServerHello(TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8_SHA256)\nCertificate*\nServerKeyExchange(ECDH PublicKey + ECC Curve Param)\nServerHelloDone()
```
RFC 7251 doesn't define TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8_SHA256 - I think because there isn't a TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8_SHA384 and there's no reason to differentiate. So I think IETF just left it off and the original text is correct.

craigpratt replied:
Here's the RFC for reference: https://tools.ietf.org/html/rfc7251

danmihai1 replied:
Thanks Craig. Pawel, can you please change TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8_SHA256 to TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8?

I guess Pawel and I used the wrong baseline file. it wasn't our intention to make this kind of change. Looks like out our TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8_SHA256 came from:

commit defb45b5849de7f1c9de2ee54e8681bee6b1f13a
Author: Craig Pratt <craig@ecaspia.com>
Date:   Fri Oct 7 18:11:43 2016 -0700

    Initial files from "v1.1 UML starting point.zip"

craigpratt replied:
Ah - that would explain it.

I thought I'd already fixed this once... ;^J

Rebasing your change onto ocf/1.0/spec is one way to fix it.

PawelWMS committed:
CR-1339: Reducing onboarding complexity between OBT and ND.

A new proposal to allow use of a single encrypted session between
OBT and ND to set additional credentials, instead of setting up
a new connection after the inital onboarding.

Signed-off-by: Pawel Winogrodzki <pawelwi@microsoft.com>

PawelWMS committed:
CR-1291: Changing Client/ServerHello.

Switching from LS_ECDHE_ECDSA_WITH_AES_128_CCM_8_SHA256 to
LS_ECDHE_ECDSA_WITH_AES_128_CCM_8.

Signed-off-by: Pawel Winogrodzki <pawelwi@microsoft.com>

PawelWMS commented:
@craigpratt: I've made the change you've asked for.

It also seems I might have messed something up with the rebase, since the request now shows the changes from CR-1339 and I'm guessing they are already pulled to ocf/1.0/spec.

craigpratt commented on security/DOXM-OwnerCreds.txt:
```diff
@@ -44,8 +44,10 @@ OT->ND: POST /oic/sec/pstat[{..., "cm:":"bx0011,1100",...}]
+
+note over OT, ND
+Provision the new device with additional credentials for device management and device-to-device communications. 
+end note
```
Actually, I think this was missing in the CR-1339 change and you must have had it in the copy committed for CR-1291. (think 1339 only modified the Just Works scenario - and I didn't notice it until now)

danmihai1 replied:
security/DOXM-OwnerCreds.txt from https://github.com/openconnectivityfoundation/uml/pull/1/files covers CR 1339. It applies to all OTMs - not Just Works. 

So, this pull request covered both CR 1339 and CR 1291.

craigpratt replied:
Well, there was a separate one for 1339 that changed this same section in the JW sequence. The one for cert OTM was swept up in this one. A little weird - but end result is fine...
